gateway/proxy_muxer.go

```diff
@@ -472,6 +472,9 @@ func (m *proxyMux) serve(gw *Gateway) {
 				WriteTimeout: writeTimeout,
 				Handler:      handler,
 			}
+			if gw.ConnectionWatcher != nil {
```

Comment:
please correct me if I'm wrong. the file handle leaks via sockets are from upstream connections right? and this mux I believe is to serve gw requests.